Easydict/NewApp/Utility/Extensions/QueryService+ConfigurableService/OpenAIService+ConfigurableService.swift

```diff
@@ -9,27 +9,65 @@
+protocol EnumLocalizedStringConvertible {
+    var title: String { get }
+}
+
+enum OpenAIModels: String, CaseIterable, Identifiable {
+    var id: Self {
+        self
+    }
+
+    case gpt3_5_turbo = "gpt-3.5-turbo"
+    case gpt3_5_turbo_1106 = "gpt-3.5-turbo-1106"
+    case gpt4
+    case gpt4_32k = "gpt-4-32k"
+}
+
+extension OpenAIModels: EnumLocalizedStringConvertible {
+    var title: String {
+        rawValue
+    }
+}
+
+@available(macOS 13.0, *)
 extension EZOpenAIService: ConfigurableService {
     func configurationListItems() -> some View {
-        ServiceStringConfigurationSection(
-            textFieldTitleKey: "service.configuration.openai.api_key.header",
-            headerTitleKey: "service.configuration.openai.api_key.title",
-            key: .openAIAPIKey,
-            prompt: "service.configuration.openai.api_key.prompt",
-            footer: {
-                Text("service.configuration.openai.api_key.footer")
-            }
-        )
+        ServiceConfigurationSecretSectionView(service: self, observeKeys: [.openAIAPIKey]) {
+            ServiceConfigurationSecureInputCell(
+                textFieldTitleKey: "service.configuration.openai.api_key.title",
+                key: .openAIAPIKey,
+                placeholder: "service.configuration.openai.api_key.placeholder"
+            )
+            // endpoint
+            ServiceConfigurationInputCell(
+                textFieldTitleKey: "service.configuration.openai.endpoint.title",
+                key: .openAIEndPoint,
+                placeholder: "service.configuration.openai.endpoint.placeholder"
+            )
+            // model
+            ServiceConfigurationPickerCell(
+                titleKey: "service.configuration.openai.model.title",
+                key: .openAIModel,
+                values: OpenAIModels.allCases
+            )
 
-        ServiceStringConfigurationSection(
-            textFieldTitleKey: "service.configuration.openai.translation.header",
-            headerTitleKey: "service.configuration.openai.translation.title",
-            key: .openAITranslation,
-            prompt: "service.configuration.openai.translation.prompt",
-            footer: {
-                Text("service.configuration.openai.translation.footer")
-            }
-        )
+            ServiceConfigurationToggleCell(
+                titleKey: "service.configuration.openai.translation.title",
+                key: .openAITranslation
+            )
+            ServiceConfigurationToggleCell(
+                titleKey: "service.configuration.openai.sentence.title",
+                key: .openAISentence
+            )
+            ServiceConfigurationToggleCell(
+                titleKey: "service.configuration.openai.dictionary.title",
+                key: .openAIDictionary
+            )
+            ServiceConfigurationToggleCell(
+                titleKey: "service.configuration.openai.usage_status.title",
+                key: .openAIServiceUsageStatus
```

Comment:
Note that `.openAIServiceUsageStatus` is enum type, we need to use Picker.

```objc
FOUNDATION_EXPORT NSString *const EZServiceUsageStatusKey;
typedef NS_ENUM(NSUInteger, EZServiceUsageStatus) {
    EZServiceUsageStatusDefault = 0,
    EZServiceUsageStatusAlwaysOff = 1,
    EZServiceUsageStatusAlwaysOn = 2,
};
```

<img width="786" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/fb5f3eec-bb31-4c7f-b408-c25add8080f1">

Reply:
OK, changed.